### ISSUE TITLE
Empty two-digit numbered-list item crashes application

### ISSUE THREAD

**Issue by sicherha** [stefan-niedermann/nextcloud-notes#668]
This is a curious bug. 🙂 The application reproducibly crashes when previewing a note that contains a numbered list, given the following two constraints:
* The number of the last list item has at least two digits.
* That last list item contains no actual text.

### Affected version
Nextcloud Notes 2.4.0 from F-Droid

### How to reproduce
1. Create a new note.
2. Insert the following text into the note (yes, that's only three characters 😉):
```md
10.
```
3. Tap the preview button (👁).
4. Witness the application go down in flames.

The above text is a minimal example. The crash is reproducible with arbitrary numbered lists as long as the aforementioned constraints are fulfilled.

### Stack trace
```
Version: 2.4.0

java.lang.StringIndexOutOfBoundsException: length=3; index=3
	at java.lang.String.charAt(Native Method)
	at com.yydcdut.markdown.syntax.text.ListSyntax.checkOrderLegal(ListSyntax.java:120)
	at com.yydcdut.markdown.syntax.text.ListSyntax.isMatch(ListSyntax.java:48)
	at com.yydcdut.markdown.chain.MultiSyntaxChain.handleSyntax(MultiSyntaxChain.java:45)
	at com.yydcdut.markdown.syntax.text.TextFactory.parseTotal(TextFactory.java:205)
	at com.yydcdut.markdown.syntax.text.TextFactory.parse(TextFactory.java:199)
	at com.yydcdut.markdown.MarkdownProcessor.parse(MarkdownProcessor.java:47)
	at it.niedermann.owncloud.notes.android.fragment.NotePreviewFragment.onActivityCreated(NotePreviewFragment.java:121)
	at android.app.Fragment.performActivityCreated(Fragment.java:2362)
	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1014)
	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1171)
	at android.app.BackStackRecord.run(BackStackRecord.java:816)
	at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1578)
	at android.app.FragmentManagerImpl$1.run(FragmentManager.java:483)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6186)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)

```

**Comment by stefan-niedermann** [comment]
Nice research man! Thank you, this will help to find a workaround (unfortunately RxMarkdown is not maintained anymore what makes *really* fixing bugs very hard).

**Comment by stefan-niedermann** [comment]
I implemented a workaround for `v2.4.3` where the app no longer crashes.

In case this error appears, the content will now be shown as usual text (markdown won't get interpreted) and a Toast hints to what the user can do to prevent this error.

We **really** have to push #549 forward :disappointed: 

Nevertheless: Thanks for reporting!
